qi-lib/on.rkt

```diff
@@ -21,28 +22,29 @@
    #`((flow clause) #,@(syntax->list #'ags))])
 
 (define-syntax-parser flow-lambda
-  [(_ (arg:id ...) expr:expr ...)
-   #'(lambda (arg ...)
-       (on (arg ...)
-           expr ...))]
-  [(_ rest-args:id expr:expr ...)
-   #'(lambda rest-args
-       (on (rest-args)
-           expr ...))])
+  [(_ args:formals clause:clause)
```

Comment:
This does make it more consistent. FTR at the moment the clause syntax class is identical to expr, but I think at some point in the past there was something different about it - I seem to recall it also kept track of the number of input arguments or something like that. It's plausible that there will be another use for it in the future (maybe for compiler optimizations??), so that's why I haven't removed it.